.github/workflows/update-styles.yml

```diff
@@ -11,9 +11,42 @@ on:
     - cron: '0 0 * * 3'
 
 jobs:
+  check-token:
+    name: "Check STYLES_WORKFLOW token"
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'carpentries/lesson-example' }}
+    outputs:
+      wf: ${{ steps.validate.outputs.wf }}
+      repo: ${{ steps.validate.outputs.repo }}
+    steps:
+      - name: "validate token"
+        id: validate
+        uses: carpentries/actions/check-valid-credentials@main
+        with:
+          token: ${{ secrets.STYLES_WORKFLOW }}
+
+  bad-token:
+    name: "Invalid/Missing Token"
+    runs-on: ubuntu-latest
+    needs: check-token
+    if: ${{ needs.check-token.outputs.wf != 'true' && github.repository == 'carpentries/lesson-example' }}
+    steps:
+      - name: "Instructions to create a new token"
+        run: |
+          printf "::warning::The STYLES_WORKFLOW secret is missing, invalid, "\
+          "or does not have the right scope to update workflows.\n"\
+          "If you want to have automated pull request updates to your workflows, "\
+          "you will need to generate a new token by visiting "\
+          "https://github.com/settings/tokens/new?scopes=repo,workflow&description=Styles%%20Token\n"\
+          "Once you have created the token, go to "\
+          "https://github.com/${{ github.repository }}/settings/secrets/actions/STYLES_WORKFLOW "\
+          "to update it."
+
   update-styles:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```

Comment:
`ubuntu-latest` will be changing over time, so I don't know if this is a good idea to use it instead of a specific version, but I'm OK with that.

Reply:
My thinking is that this workflow relies only on BASH, so updates to Ubuntu should not affect it (as opposed to other workflows that depended on a number of apt libraries)